test/Concurrency/await_typo_correction.swift

```diff
@@ -0,0 +1,31 @@
+// RUN: %target-typecheck-verify-swift -enable-experimental-concurrency -parse-as-library
+// REQUIRES: concurrency
+
+func asyncFunc() async throws {}
+
+func anotherAsyncFunc() async -> Int {
+  return 42
+}
+
+@main struct MyProgram {
+  static func main() async throws {
+    // expected-error@+7 {{cannot find 'async' in scope; did you mean 'await'?}}{{9-14=await}}
+    // expected-error@+6 {{consecutive statements on a line must be separated by ';'}}
+    // expected-error@+5 {{call is 'async' but is not marked with 'await'}}
+    // expected-error@+4 {{call can throw but is not marked with 'try'}}
+    // expected-note@+3 {{did you mean to use 'try'?}}
+    // expected-note@+2 {{did you mean to handle error as optional value?}}
+    // expected-note@+1 {{did you mean to disable error propagation?}}
+    try async asyncFunc()
```

Comment:
Couple of things:

1. IMO `cannot find 'async' in scope; did you mean 'await'?` is the wrong diagnostic message. Can we instead simplify it to (say) `found 'async' in expression, did you mean 'await'?` ? I understand that technically `async` is a contextual keyword, but we should focus on the 95% use case, where someone is very likely referring to the `async` keyword.
2. [Not strictly part of this PR but] Can we avoid a cascade of errors here? It seems really unfortunate to have 7 (!) messages all at the same spot. Despite being a compiler developer, I have no idea what the "did you mean to disable error propagation?" means...

Reply:
(For the error propagation one, I see there is a fix-it to add `try!`, so it's more understandable with the fix-it, but I still think the number of messages is too high here. For example, we shouldn't be saying "did you mean to use 'try'?" if there is a `try` already present.)

Reply:
Okay, now we only emit the one error about replacing the `async` with `await.